enhancements/network/networking-console-plugin.md

```diff
@@ -0,0 +1,148 @@
+---
+title: networking-console-plugin
+authors:
+  - "@upalatucci"
+reviewers:
+  - "@orenc1"
+  - "@zshi-redhat"
+  - "@pcbailey"
+  - "@metalice"
+  - "@spadgett"
+  - "@jhadvig"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-04-12
+last-updated: 2024-04-12
+tracking-link:
+  - https://issues.redhat.com/browse/CONSOLE-3952
+---
+
+
+# Create console plugin for the networking section
+
+## Summary
+
+At present, several pages within the networking section of our console are defined within the console repository. 
+However, working with the console repository has slowed down development and there are multiple motivations to convert static console plugins into dynamic ones described [here](https://github.com/spadgett/enhancements/blob/master/enhancements/console/dynamic-plugins.md#motivation). 
+To address this, we propose relocating the relevant code to a separate repository named "networking-console-plugin" and integrating it into the cluster-networking-operator.
+Specifically: Services, Route, Ingress, NetworkPolicy, and NetworkAttachmentDefinition pages for listing, creating, and editing those resources.
+
+
+
+
+## Motivation
+
+### User Stories
+
+N/A. For the end user the UI will be the same. The cluster admin will have the option to disable and enable the plugin but it's not relevant. 
```

Comment:
```suggestion
N/A. For the end user the UI will be the same. The cluster admin will have the option to disable and enable the plugin but it's not relevant. 
```
I'm interested in this disable option, can you please expand more?

Reply:
All the plugins installed in the cluster can be listed using this command

`oc get consoleplugins`

You can verify which plugins are enabled using this command

`oc get console.operator.openshift.io cluster -o jsonpath='{.spec.plugins}'`

Disabling plugins means that the pages and features they provide will not be shown in the console. 

For example, this plugin defined an entire section in the console called 'Networking'. Disabling this plugin means that the user can no longer see this section in the console UI.